typing_extensions/src/typing_extensions.py

```diff
@@ -2048,40 +2048,61 @@ def __eq__(self, other):
 
     TypeGuard = _TypeGuard(_root=True)
 
+if not hasattr(typing, "Self") and sys.version_info[:2] >= (3, 7):
+    # Vendored from cpython typing._SpecialFrom
```

Comment:
It's a bit hard to follow the if chains here, but can't this just be together with the appropriate branch for 3.7-3.10 so we don't have to duplicate the conditions?

Reply:
Should I refactor the uses of typing._SpecialForm to use this vendored  verion too?

Reply:
Let's change Self only for simplicity.